Added MLSToSlopes

JanWehrmann opened the pull request:
MLSToSlopes in /tools is a class used to compute slopes
and maxSteps from a MLSMapKalman.
Tests were added and the necessary changes to the CMakeLists
were made.

saarnold commented on src/tools/MLSToSlopes.hpp:
```diff
@@ -0,0 +1,93 @@
+     * have one patch per cell.
+     *
+     * It can be customized by subclassing and overloading the computeGradient
+     * operator
```
If you want support this the computeGradient method must be virtual.
It seems this method is not implemented or used?

JanWehrmann replied:
Yes. That was also the case in the original envire version.

saarnold replied:
Lets remove this method and the comment then. It was probably replaced by the PlaneFitting

saarnold commented on src/tools/MLSToSlopes.hpp:
```diff
@@ -0,0 +1,93 @@
+				
+        double computeGradient(double mean0, double mean1, double stdev0, double stdev1);
+		
+		/**
```
You mixed spaces and tabs. Please use only spaces.

saarnold commented on src/tools/MLSToSlopes.hpp:
```diff
@@ -0,0 +1,93 @@
+		* The output GridMaps will be fitted to match the size and resolution of the input.
+		* If not enough information is available for a cell all ouput values are set to -inf.
+		* */
+		bool convertToSlopes(maps::grid::MLSMapKalman input, maps::grid::GridMapF& slopesOut,
```
The input parameter should be a const reference, otherwise a copy will be made.

saarnold commented on src/tools/MLSToSlopes.hpp:
```diff
@@ -0,0 +1,88 @@
+        * @param correctSteps :  Set to true to compute corrected max steps instead.
+        * @param correctedStepThreshold : If the value for a max step falls under the threshold, it will be corrected.
+        * */
+        static bool computeMaxSteps(maps::grid::MLSMapKalman& mlsIn, maps::grid::GridMapF& maxStepsOut,
```
All references that are not modified by the method should be const

Commit:
Added MLSToSlopes.

MLSToSlopes in /tools is a class used to compute slopes
and maxSteps from a MLSMapKalman.
Tests were added and the necessary changes to the CMakeLists
were made.

saarnold commented:
Thanks :+1:
